src/black/strings.py

```diff
@@ -137,6 +137,10 @@ def normalize_string_prefix(s: str, remove_u_prefix: bool = False) -> str:
     assert match is not None, f"failed to match string {s!r}"
     orig_prefix = match.group(1)
     new_prefix = orig_prefix.replace("F", "f").replace("B", "b").replace("U", "u")
+
+    # Python syntax guarantees max 2 prefixes and that one of them is "r"
+    if len(new_prefix) == 2 and "r" == new_prefix[0].lower():
```

Comment:
And technically the first condition is redundant, but I like it because it clarifies the meaning.

Reply:
Also it short circuits the (likely) more complex subscript and `.lower`.